### PR TITLE
Fw 4268 implement dictionary search with parameters

### PR DESCRIPTION
**Pull request by sahil97**
### Description of Changes
dictionary-search api added at `/api/1.0/sites/{{site-slug}}/dictionary-search/` with the following parameters
1. category - Accepts id for a category filter. Returns all the entries associated with the given category or its child categories.
2. startsWithChar - Accepts a char (or combination of chars) and returns entries which has the given char at the beginning.

### Checklist
- [x] README / documentation has been updated if needed
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- ~Admin Panel has been updated if models have been added or modified~
- ~Migrations have been updated and committed if applicable~
- [x] Team Postman workspace has been updated if applicable

### Reviewers Should Do The Following:
- [x] Review the code changes here
- ~Pull the branch and test locally~

### Additional Notes
N/A

### CODE REVIEW

File: firstvoices/backend/tests/test_search/test_query_params.py [hunk 1]
```diff
@@ -152,3 +153,100 @@ def test_both(self):
...
+
+    def test_default(self):  # default case
+        # relates to: SearchQueryTest.java - CategoryIdTest.testDefault(),
+        # CategoryIdTest.test_category_blank_allowed()
+        search_query = get_search_query()
+        search_query = search_query.to_dict()
+
+        assert "categories" not in str(search_query)
+
+    def test_invalid_category(self):
```
Comment (sarahfirstvoices) [inline firstvoices/backend/tests/test_search/test_query_params.py:175]: You can just remove this if it isn't testing anything. Maybe move this comment onto the TestCategory class level

File: firstvoices/backend/urls.py [hunk 2]
```diff
@@ -53,6 +54,9 @@
 sites_router.register(r"images", ImageViewSet, basename="image")
 sites_router.register(r"people", PersonViewSet, basename="person")
 sites_router.register(r"search", SiteSearchViewsSet, basename="site-search")
+sites_router.register(
+    r"dictionary-search", DictionarySearchViewSet, basename="dictionary-search"
```
Comment (sarahfirstvoices) [inline firstvoices/backend/urls.py:58]: Any reason not to just have this be the list view of /dictionary like it was for the old api?

File: firstvoices/backend/search/utils/object_utils.py [hunk 3]
```diff
@@ -108,3 +108,11 @@ def get_notes_text(dictionary_entry_instance):
     for note in notes_set:
         notes_text.append(note.text)
     return " ".join(notes_text)
+
+
+def get_categories_ids(dictionary_entry_instance):
+    categories_set = dictionary_entry_instance.categories.all()
+    categories_list = []
+    for category in categories_set:
```
[the remark this answers is not in the document]
Comment (sahil97) [reply firstvoices/backend/search/utils/object_utils.py:116]: Yes, I forgot about the function even though I did use it in a different place. Thanks.

File: firstvoices/backend/search/utils/query_builder_utils.py [hunk 4]
```diff
@@ -197,6 +232,27 @@ def get_valid_domain(input_domain_str):
         return None
 
 
+def get_valid_starts_with_char(input_str):
+    # Starting alphabet can be a combination of characters as well
+    # taking only first word if multiple words are supplied
+    valid_str = str(input_str).strip().lower().split(" ")[0]
+    return valid_str
```
Comment (bronson-c) [inline firstvoices/backend/search/utils/query_builder_utils.py:239]: While you're sanitizing input you could probably clean confusables with the `clean_confusables` function on the alphabet model here as well.

File: firstvoices/backend/urls.py (hunk 2 above)
Comment (bronson-c) [reply firstvoices/backend/urls.py:58]: +1 to this question

**Review by sarahfirstvoices — APPROVED** [review]

File: firstvoices/backend/urls.py (hunk 2 above)
Comment (bronson-c) [reply firstvoices/backend/urls.py:58]: +1 for /dictionary/search as well, but yes it's totally just stylistic

**Review by bronson-c — APPROVED** [review]
Looks good, thanks for the addition!

**Pull request merged**